src/main/java/seedu/address/model/assignment/DeadlineContainsKeywordsPredicate.java

```diff
@@ -0,0 +1,32 @@
+package seedu.address.model.assignment;
+
+import java.util.List;
+import java.util.function.Predicate;
+
+import seedu.address.commons.util.StringUtil;
+
+/**
+ * Tests that a {@code Assignment}'s {@code Deadline} matches any of the keywords given.
+ */
+public class DeadlineContainsKeywordsPredicate implements Predicate<Assignment> {
+    private final List<String> keywords;
+
+    public DeadlineContainsKeywordsPredicate(List<String> keywords) {
+        this.keywords = keywords;
+    }
+
+    @Override
+    public boolean test(Assignment assignment) {
+        return keywords.stream()
+                .anyMatch(keyword -> StringUtil
+                        .containsWordIgnoreCase(assignment.getDeadline().value, keyword));
+    }
```

Comment:
According to the CS2103T coding conventions, boolean methods and boolean variables should be named to sound like a boolean. However, it appears that this method is overriding a method from Predicate, so it should be fine :)

Reply:
:D